tests/unit/runners/test_bgp.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+# Import Python Libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import salt libs
+import salt.runners.bgp as bgp
+
+# Import Salt Testing Libs
+from tests.support.mixins import LoaderModuleMockMixin
+from tests.support.unit import TestCase, skipIf
+
+
+@skipIf(not bgp.HAS_NAPALM, "napalm module required for this test")
+class BGPTest(TestCase, LoaderModuleMockMixin):
+    """
+    Test the bgp runner
+    """
+
+    def setup_loader_modules(self):
+        return {
+            bgp: {
+                "__opts__": {
+                    "optimization_order": [0, 1, 2],
+                    "renderer": "yaml",
+                    "renderer_blacklist": [],
+                    "renderer_whitelist": [],
+                }
+            }
+        }
+
+    def test_neighbors(self):
+        ret = bgp.neighbors()
+        self.assertEqual(ret, [])
```

Comment:
Before anything, thank you @Akm0d for helping out here. (the following comments are not aiming you or any individual, rather SaltStack's views on this community and contributors)

I'd like however to raise some questions, just for my understanding of how this process works: what does this test? What specific purpose does it serve? If I were to submit a PR now that entirely changes the behaviour of this function, would this test catch it?
I could be wrong, but I'm seeing that it doesn't test anything at all, it barely just confirms the existence of a function. So have we blocked this PR for over 6 months for tests that don't test anything?

Note, I'm not questioning the importance of the tests, but the rule itself: I understand that everything needs a test, I get that; but it's disappointing the way this rule is followed: blindly, robotically, asking contributors to toss in tests when just making a small adjustment (asking to write tests from scratch, when there isn't any) - but everyone's completely happy when throwing some non-sense tests, right? That's just silly. It may make sense to SS as "hey, our code is X% tested", but to contributors like me, it doesn't, it only makes me feel more and more loathe to contribute to this project, I'm disappointed - to put it mildly.

If SS would ever gave a damn about the community as it's lately started trumpeting left and right, would think these decisions thoroughly, also from the perspective of this community, but sadly, it doesn't.

Reply:
For what it's worth, we realized that many of our contributors are inexperienced with writing tests, so I've started a Test Clinic, streaming live on Tuesday afternoon and Tuesday mornings (Central/US Time zones).

We have been encouraging contributors to come and learn more about how to write *meaningful* tests, and especially bring their PRs that need tests, and I'm happy to walk through what tests would actually be meaningful, and even start writing tests.

---

When it comes to blindly following this rule, one thing that has stuck out to me is that code missing tests have *significantly* more bugs, in some cases an entire state just wouldn't even work because they're relying on functions or function signatures that literally don't even exist in the module!

And this is code that had been merged into other branches :grimacing:.

Based on this experience, I will take code with blindly added tests that just check that a function even exists, over code without tests, any day. Granted, I'd prefer they have *meaningful* test coverage -- and I'm more than happy to discuss how to achieve that for any PR, during the test clinic, on IRC, Slack, or anywhere else.

---

We absolutely do want meaningful tests, and we realize that's difficult, and are doing as much as we can to support writing tests.

If there's something that you see that we *could* be doing, or doing better, please do bring that to our attention :pray:. Besides the Test Clinic, is there anything we could be doing better to support contributors in writing tests for contributions?